bundle/EventListener/QueryTypeListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Netgen\Bundle\EzPlatformSiteApiBundle\EventListener;
+
+use eZ\Publish\Core\MVC\Symfony\Event\PreContentViewEvent;
+use eZ\Publish\Core\MVC\Symfony\MVCEvents;
+use Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * QueryTypeListener processes query type configuration, contained in the View parameters.
+ * Result of processing is a QueryCollection, which will be assigned to the View parameter.
+ */
+class QueryTypeListener implements EventSubscriberInterface
+{
+    /**
+     * @var \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper
+     */
+    private $queryCollectionMapper;
+
+    /**
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper $queryCollectionMapper
+     */
+    public function __construct(QueryCollectionMapper $queryCollectionMapper)
+    {
+        $this->queryCollectionMapper = $queryCollectionMapper;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            MVCEvents::PRE_CONTENT_VIEW => [
```

Comment:
You should listen to `eZ\Publish\Core\MVC\Symfony\View\ViewEvents::FILTER_VIEW_PARAMETERS` event, which exists for this exact purpose.

See here for examples: https://github.com/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/Core/MVC/Symfony/View/ParametersInjector

Reply:
Removed in https://github.com/netgen/ezplatform-site-api/pull/70/commits/04f87dc1ccbcaa223eb6826dbcca6e9c829bef78